asset-transfer-ledger-queries/chaincode-go/asset_transfer_ledger_chaincode.go

```diff
@@ -2,89 +2,93 @@
+
+This asset transfer ledger example chaincode demonstrates a packaged
+index which you can find in META-INF/statedb/couchdb/indexes/indexOwner.json.
+
+If you have access to the your peer's CouchDB state database in a development environment,
+you may want to iteratively test various indexes in support of your chaincode queries.  You
+can use the CouchDB Fauxton interface or a command line curl utility to create and update
+indexes. Then once you finalize an index, include the index definition alongside your
+chaincode in the META-INF/statedb/couchdb/indexes directory, for packaging and deployment
+to managed environments.
+
+In the examples below you can find index definitions that support asset transfer ledger
+chaincode queries, along with the syntax that you can use in development environments
+to create the indexes in the CouchDB Fauxton interface or a curl command line utility.
+
+
+Index for docType, owner.
+
+Example curl command line to define index in the CouchDB channel_chaincode database
+curl -i -X POST -H "Content-Type: application/json" -d "{\"index\":{\"fields\":[\"docType\",\"owner\"]},\"name\":\"indexOwner\",\"ddoc\":\"indexOwnerDoc\",\"type\":\"json\"}" http://hostname:port/myc1_assets/_index
+
+
+Index for docType, owner, size (descending order).
+
+Example curl command line to define index in the CouchDB channel_chaincode database:
+curl -i -X POST -H "Content-Type: application/json" -d "{\"index\":{\"fields\":[{\"size\":\"desc\"},{\"docType\":\"desc\"},{\"owner\":\"desc\"}]},\"ddoc\":\"indexSizeSortDoc\", \"name\":\"indexSizeSortDesc\",\"type\":\"json\"}" http://hostname:port/myc1_assets/_index
+
+Rich Query with index design doc and index name specified (Only supported if CouchDB is used as state database):
+peer chaincode query -C myc1 -n asset_transfer -c '{"Args":["QueryAssets","{\"selector\":{\"docType\":\"asset\",\"owner\":\"tom\"}, \"use_index\":[\"_design/indexOwnerDoc\", \"indexOwner\"]}"]}'
+
+Rich Query with index design doc specified only (Only supported if CouchDB is used as state database):
+peer chaincode query -C myc1 -n asset_transfer -c '{"Args":["QueryAssets","{\"selector\":{\"docType\":{\"$eq\":\"asset\"},\"owner\":{\"$eq\":\"tom\"},\"size\":{\"$gt\":0}},\"fields\":[\"docType\",\"owner\",\"size\"],\"sort\":[{\"size\":\"desc\"}],\"use_index\":\"_design/indexSizeSortDoc\"}"]}'
+*/
 
 package main
 
 import (
 	"encoding/json"
 	"fmt"
+	"log"
 	"strings"
 	"time"
 
+	"github.com/golang/protobuf/ptypes"
 	"github.com/hyperledger/fabric-chaincode-go/shim"
 	"github.com/hyperledger/fabric-contract-api-go/contractapi"
 )
 
-// SimpleChaincode example simple Chaincode implementation
+// SimpleChaincode implements the fabric-contract-api-go programming model
 type SimpleChaincode struct {
 	contractapi.Contract
 }
 
 type asset struct {
 	ObjectType     string `json:"docType"` //docType is used to distinguish the various types of objects in state database
```

Comment:
vote for a consistent terminology between ObjectType & docType: How about  Type or DocType ?
 
I am working on private-data samples, so we can make the Asset consistent as much as possible.

To avoid changing queries:
```suggestion
	Type     string `json:"docType"` //docType is used to distinguish the various types of objects in state database
```

Reply:
Agreed, lets go with `DocType` as lowercase `Type` (type) is a reserved word. In general there is little reason for fields to not match their tag names

Reply:
Done